ospf6d/ospf6_asbr.c

```diff
@@ -1866,36 +1901,70 @@ static void ospf6_asbr_external_route_show(struct vty *vty,
 		snprintf(forwarding, sizeof(forwarding), ":: (ifindex %d)",
 			 ospf6_route_get_first_nh_index(route));
 
-	vty_out(vty, "%c %-32pFX %-15s type-%d %5lu %s\n",
-		zebra_route_char(info->type), &route->prefix, id,
-		route->path.metric_type,
-		(unsigned long)(route->path.metric_type == 2
-					? route->path.u.cost_e2
-					: route->path.cost),
-		forwarding);
+	if (use_json) {
+		json_route = json_object_new_object();
+		snprintf(route_type, sizeof(route_type), "%c",
+			 zebra_route_char(info->type));
+		json_object_string_add(json_route, "routeType", route_type);
+		json_object_string_add(json_route, "destination", prefix);
+		json_object_string_add(json_route, "id", id);
+		json_object_int_add(json_route, "metricType",
+				    route->path.metric_type);
+		json_object_int_add(
+			json_route, "routeCost",
+			(unsigned long)(route->path.metric_type == 2
+						? route->path.u.cost_e2
+						: route->path.cost));
+		json_object_string_add(json_route, "forwarding", forwarding);
+
+		json_object_array_add(json_array, json_route);
+	} else
+
+		vty_out(vty, "%c %-32pFX %-15s type-%d %5lu %s\n",
+			zebra_route_char(info->type), &route->prefix, id,
+			route->path.metric_type,
+			(unsigned long)(route->path.metric_type == 2
+						? route->path.u.cost_e2
+						: route->path.cost),
+			forwarding);
 }
 
-DEFUN (show_ipv6_ospf6_redistribute,
```

Comment:
Let's put this defun back to the same formatting as before.  I know clang-format fucks it up, but I would prefer we keep the formatting the same here for the rest of the file.

Reply:
resolved